engine-tests/src/tests/pause_contract.rs

```diff
@@ -0,0 +1,161 @@
+use crate::test_utils;
+use aurora_engine::parameters::SetUpgradeDelayBlocksArgs;
+use borsh::BorshSerialize;
+
+#[test]
+fn test_pause_contract_require_owner() {
+    let mut runner = test_utils::deploy_evm();
+    let aurora_account_id = runner.aurora_account_id.clone();
+
+    let result = runner.call("pause_contract", &aurora_account_id, vec![]);
+    assert!(result.is_ok());
+
+    let result = runner.call("resume_contract", &aurora_account_id, vec![]);
+    assert!(result.is_ok());
+
+    let result = runner.call("pause_contract", "new_owner.near", vec![]);
+    assert!(result.is_err());
+}
+
+#[test]
+fn test_resume_contract_require_owner() {
+    let mut runner = test_utils::deploy_evm();
+    let aurora_account_id = runner.aurora_account_id.clone();
+
+    let result = runner.call("pause_contract", &aurora_account_id, vec![]);
+    assert!(result.is_ok());
+
+    let result = runner.call("resume_contract", &aurora_account_id, vec![]);
+    assert!(result.is_ok());
+
+    let result = runner.call("pause_contract", &aurora_account_id, vec![]);
+    assert!(result.is_ok());
+
+    let result = runner.call("resume_contract", "new_owner.near", vec![]);
+    assert!(result.is_err());
+}
```

Comment:
I would decrease the number of calls in the above tests. Maybe even combine them. The reason is that every call takes about 1 - 2 seconds.

Reply:
Yeah, its was to much. I concentrated the scenarios in fewer tests.